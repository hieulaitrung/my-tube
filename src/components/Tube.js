import Avatar from '@material-ui/core/Avatar';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import { red } from '@material-ui/core/colors';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import { makeStyles } from '@material-ui/core/styles';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ShareIcon from '@material-ui/icons/Share';
import React from 'react';

const useStyles = makeStyles((theme) => ({
    root: {
        width: 350,
        height: 330
    },
    media: {
        height: 0,
        paddingTop: '56.25%', // 16:9
    },
    expand: {
        transform: 'rotate(0deg)',
        marginLeft: 'auto',
        transition: theme.transitions.create('transform', {
            duration: theme.transitions.duration.shortest,
        }),
    },
    expandOpen: {
        transform: 'rotate(180deg)',
    },
    avatar: {
        backgroundColor: red[500],
    },
}));

const Tube = (props) => {
    const tube = props.item;
    const classes = useStyles();
    const authorFirstName = tube.author.email.charAt(0);
    const dateDisplay = new Date(tube.date).toDateString();
    return (
        <Grid item>
            <Card className={classes.root}>
            <CardHeader
                avatar={
                    <Avatar aria-label="recipe" className={classes.avatar}>
                        {authorFirstName}
          </Avatar>
                }
                title={tube.title}
                subheader={dateDisplay}
            />
            <CardMedia
                className={classes.media}
                image="https://via.placeholder.com/150"
                title={tube.title}
            />
            <CardActions disableSpacing>
                <IconButton aria-label="add to favorites">
                    <FavoriteIcon />
                </IconButton>
                <IconButton aria-label="share">
                    <ShareIcon />
                </IconButton>
            </CardActions>
        </Card>
        </Grid>
    )
}

export default Tube
