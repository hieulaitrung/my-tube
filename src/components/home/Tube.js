import Avatar from '@material-ui/core/Avatar';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Link from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';
import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import TubeThumbnail from '../TubeThumbnail';

const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        border: 'none',
        background: 'transparent',
    },
    link: {
        '&:hover': {
            'text-decoration': 'none',
        }
    },
    media: {
        height: 0,
        paddingTop: '56.25%', // 16:9
    },
    content: {
        paddingTop: theme.spacing(1.5),
        [theme.breakpoints.up("xs")]: {
            paddingLeft: 0
        },
        [theme.breakpoints.down("xs")]: {
            paddingLeft: theme.spacing(1.5)
        }
    },
    header: {
        padding: 0,
        alignItems: 'flex-start'
    },

}));

const Tube = (props) => {
    const tube = props.item;
    const classes = useStyles();
    const authorFirstName = tube.author.email.charAt(0);
    const dateDisplay = new Date(tube.date._seconds * 1000).toDateString();
    return (
        <Link className={classes.link} component={RouterLink} to={`/watch?v=${tube.id}`}>
            <Card className={classes.root} variant="outlined" square={true}>
                <TubeThumbnail className={classes.media} tube={tube}/>
                <CardContent className={classes.content}>
                    <CardHeader
                        avatar={
                            <Avatar aria-label="recipe">
                                {authorFirstName}
                            </Avatar>
                        }
                        title={tube.title}
                        subheader={dateDisplay}
                        className={classes.header}
                    />
                </CardContent>
            </Card>
        </Link>
    )
}

export default Tube
