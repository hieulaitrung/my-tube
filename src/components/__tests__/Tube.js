import React from 'react';
import Tube from '../home/Tube';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';
import Link from '@material-ui/core/Link';

const tube = {
    id: 1,
    title: 'Title',
    body: 'Body',
    author: {
        email: 'example@fake.com'
    },
    date: 1
};

describe('<Tube />', () => {
    test('snapshot renders', () => {
        
        const component = renderer.create(
            <Router>
                <Tube item={tube} />
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders a <Link/> component', () => {
        const wrapper = mount(
            <Router>
                <Tube item={tube} />
            </Router>
        );
        const links =  wrapper.find(Link);
        expect(links.length).toEqual(1);
        expect(links.get(0).props.to).toEqual(`/watch?v=${tube.id}`);
    });
    
    it('should renders a <CardHeader/> component', () => {
        const dateDisplay = new Date(tube.date).toDateString();
        const wrapper = mount(
            <Router>
                <Tube item={tube} />
            </Router>
        );
        const cardHeaders =  wrapper.find(CardHeader);
        expect(cardHeaders.length).toEqual(1);

        const cardHeader= cardHeaders.get(0);
        expect(cardHeader.props.title).toEqual(tube.title);
        expect(cardHeader.props.subheader).toEqual(dateDisplay);

        expect(cardHeaders.find(Avatar).length).toEqual(1);
    });

});