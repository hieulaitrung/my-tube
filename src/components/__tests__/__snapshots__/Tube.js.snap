// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tube /> snapshot renders 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-2 MuiTypography-colorPrimary"
  href="/watch?v=1"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined"
  >
    <div
      className="MuiCardMedia-root makeStyles-media-3"
      style={
        Object {
          "backgroundImage": "url(\\"https://via.placeholder.com/256\\")",
        }
      }
      title="Title"
    />
    <div
      className="MuiCardContent-root makeStyles-content-4"
    >
      <div
        className="MuiCardHeader-root makeStyles-header-5"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <div
            aria-label="recipe"
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            e
          </div>
        </div>
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
          >
            Title
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            Invalid Date
          </span>
        </div>
      </div>
    </div>
  </div>
</a>
`;
